src/python/foglamp/core/server.py

```diff
@@ -59,12 +129,36 @@ def start(cls):
                 signal_name,
                 lambda: asyncio.ensure_future(cls._stop(loop)))
 
+        # TODO: make management port dynamic?!
+        host_address = "localhost"  # fix?!
+        core_mgt_port = 8082
+
+        # start scheduler
         # The scheduler must start first because the REST API interacts with it
         loop.run_until_complete(asyncio.ensure_future(cls._start_scheduler()))
 
-        # https://aiohttp.readthedocs.io/en/stable/_modules/aiohttp/web.html#run_app
-        web.run_app(cls._make_app(), host='0.0.0.0', port=8082, handle_signals=False)
+        loop.create_task(cls.start_storage(loop, host_address, core_mgt_port))
+        # loop.call_soon(asyncio.ensure_future(cls._start_storage(host_address, core_mgt_port)))
+
+        cls._start_core(host=host_address, management_port=core_mgt_port)
```

Comment:
This has to happen first.

Reply:
Please check via `python -m foglamp.core` or see syslog. Its being started first. 
I did not fix the scheduler line as it had comments as "MUST start before"; I'm not sure if it was designed per that architecture. we can simply move that to after storage (which I did not intentionally put yet) as `loop.create_task(cls._start_scheduler())` just next task after the `start storage` to be started in event loop ; loop  starts with start_core hence first thing.

Reply:
I suspect the way this works is that the create_task does not actually run something immediately, but rather some time later its the event loop. Of course this is dangerous if we change some in the start_core that ends up executing the event loop.